pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentObserver.java

```diff
@@ -114,8 +115,8 @@ public void onError(Throwable t) {
     LOGGER.warn("Error on receiver side", t);
     if (_mailbox != null) {
       _mailbox.setErrorBlock(
-          ErrorMseBlock.fromException(new RuntimeException("Cancelled by sender", t)),
-          Collections.emptyList());
+          ErrorMseBlock.fromException(new QueryCancelledException(
```

Comment:
A factory method like ErrorMseBlock.fromErrMsg(ErrorCode, String) would be useful here as well

Reply:
I agree, using the same strategy as in `GrpcSendingMailbox`.